### PR TITLE
add commander.js-based wrapper for scaffolding new WooCommerce extensions

### LINKED ISSUE #36575: Create Extension: Call @wordpress/create-block under the hood
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Currently, Create Extension is called via

```
npx @wordpress/create-block -t @woocommerce/create-woo-extension my-extension-name
```
It needs to be shortened to 

```
npx @woocommerce/create-woo-extension my-extension-name
```
Use Commander to wrap the call to `@wordpress/create-block`. The motivation for doing this is:

1. Provide an opportunity to make async calls, see https://github.com/woocommerce/woocommerce/pull/35922 for use cases. That PR will likely be abandoned due to no movement from Gutenberg.
2. Shorten the invoking command
3. Provide interactivity so users can craft the resulting repo depending on the example they'd like to create. See https://github.com/woocommerce/woocommerce/pull/36489 for an example.

### PR DESCRIPTION
**Pull request by lsinger**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #36575.

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1.
2.
3.

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [ ] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by psealock** [comment]
@lsinger, there's a bit going on here:

* I think there is room for an `@woocommerce/cli` package but I don't think we've thoroughly thought through what that means or entails. As a result, renaming the `@woocommerce/create-woo-extension` isn't in the scope, but interesting idea I'd love to explore in the future.
* I don't think we're ready to delete `/tools/create-extension` just yet. That is the plan but its currently referenced by a few dev blog posts. I'd like to get `@woocommerce/create-woo-extension` to a better place first.
* The folder re-structure is a cool possibility, but I'm also wondering if its adding complexity at the moment.

The desired result here goes something like this:

With the following changes you can run the package locally via `npx ../path/to/woocommerce/packages/js/create-woo-extension`

````diff
diff --git a/packages/js/create-woo-extension/index.js b/packages/js/create-woo-extension/index.js
old mode 100644
new mode 100755
index 088a483570..f6fca80659
--- a/packages/js/create-woo-extension/index.js
+++ b/packages/js/create-woo-extension/index.js
@@ -1,21 +1,2 @@
-module.exports = {
-       templatesPath: __dirname,
-       defaultValues: {
-               npmDependencies: [
-                       '@wordpress/hooks',
-                       '@wordpress/i18n',
-                       '@woocommerce/components',
-               ],
-               npmDevDependencies: [
-                       '@woocommerce/dependency-extraction-webpack-plugin',
-                       '@woocommerce/eslint-plugin',
-                       '@wordpress/prettier-config',
-                       '@wordpress/scripts@24.6.0',
-               ],
-               namespace: 'extension',
-               license: 'GPL-3.0+',
-               customScripts: {
-                       postinstall: 'composer install',
-               },
-       },
-};
+#!/usr/bin/env node
+console.log( 'hello world' );
diff --git a/packages/js/create-woo-extension/package.json b/packages/js/create-woo-extension/package.json
index 1134685729..cdd9c48710 100644
--- a/packages/js/create-woo-extension/package.json
+++ b/packages/js/create-woo-extension/package.json
@@ -3,6 +3,9 @@
        "version": "1.0.2",
        "description": "A template to be used with `@wordpress/create-block` to create a WooCommerce extension.",
        "main": "index.js",
+       "bin": {
+               "create-woo-extension": "./index.js"
+       },
        "engines": {
                "node": "^16.13.1",
                "pnpm": "^7.13.3"

**Comment by psealock** [comment]
_I needed a new comment due to formatting_

The result of `npx ../path/to/woocommerce/packages/js/create-woo-extension` will be "hello world" logged out.

now `/index.js` will house your Commander script, as you've done, but will call the `@wordpress/create-block` command which will need to somehow pass in the proper configuration (all the deleted data in `index.js`) to create a new repo.

We can restructure the folders to allow for different templates at a later date.

**Comment by lsinger** [comment]
Thanks @psealock, I definitely got carried away a bit. 😬 I simplified things again, only keeping the separation between the actual CLI tool and its meta data and the template. It was previously not very visible what belongs to the template and what belongs to the project, so keeping the template separate IMHO still makes a lot of sense.

File: packages/js/create-woo-extension/index.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-			postinstall: 'composer install',
-		},
-	},
-};
+#!/usr/bin/env node
+const path = require('node:path');
+const { Command } = require('commander');
+const slug = require('slug');
+const spawn = require('child_process').spawn;
+const package_info = require('./lib/package-info');
```
Comment (psealock) [inline packages/js/create-woo-extension/index.js:6]: JS tends to favor camelCase, ie `packageInfo`. We should stick to this convention here also

File: packages/js/create-woo-extension/template/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			license: 'GPL-3.0+',
+			attributes: {
+				latestWooVersion,
+			},
+		},
+		customScripts: {
+			postinstall: 'composer install',
+		},
+	};
+} )();
```
Comment (psealock) [inline packages/js/create-woo-extension/template/index.js:43]: Do you have Prettier setup in your IDE? I believe it takes care of these when configured to use the Prettier found in the Monorepo's dependencies.

File: packages/js/create-woo-extension/template/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	"author": "",
+	"license": "GPL-3.0+",
+	"bugs": {
+		"url": "https://github.com/woocommerce/woocommerce/issues"
+	},
+	"homepage": "https://github.com/woocommerce/woocommerce/tree/trunk/packages/js/create-woo-extensionv#readme",
+	"dependencies": {
+		"node-fetch": "^3.3.0"
+	}
+}
```
Comment (psealock) [inline packages/js/create-woo-extension/template/package.json:28]: This `package.json` seems not needed, is there a reason for its existence?

File: packages/js/create-woo-extension/package.json [hunk 4]
```diff
@@ -15,11 +18,40 @@
...
+		"wordpress"
+	],
+	"author": "Automattic",
 	"license": "GPL-3.0+",
 	"bugs": {
 		"url": "https://github.com/woocommerce/woocommerce/issues"
 	},
-	"homepage": "https://github.com/woocommerce/woocommerce/tree/trunk/packages/js/create-woo-extensionv#readme"
+	"homepage": "https://github.com/woocommerce/woocommerce/tree/trunk/packages/js/create-woo-extension#readme",
+	"require-dev": {
```
Comment (psealock) [inline packages/js/create-woo-extension/package.json:32]: The changelogger items belong in `composer.json`

File: packages/js/create-woo-extension/template/index.js (hunk 2 above)
Comment (lsinger) [reply packages/js/create-woo-extension/template/index.js:43]: I had not -- have set it up now, thanks for the reminder. 👍

File: packages/js/create-woo-extension/index.js (hunk 1 above)
Comment (lsinger) [reply packages/js/create-woo-extension/index.js:6]: Right, thanks for the catch. :ok_hand:

**Comment by lsinger** [comment]
Closing for now as priorities have changed.

**Pull request closed without merging**